### ISSUE TITLE
multi-tenant: Evaluate whether or not we can ship 22.1 with cockroach demo defaulting to multi-tenant mode

### ISSUE THREAD

**Issue by ajstorm** [cockroachdb/cockroach#72341]
#71988 enabled multi-tenant mode by default for `cockroach demo`.  When this was done, there were still some known gaps in multi-tenant demo which would prevent us from shipping it as the default mode in 22.1.  Towards the end fo the 22.1 cycle we should reevaluate the list of addressed issues to determine if we're comfortable shipping demo in MT mode as the default.

**Comment by ajstorm** [comment]
It's looking quite likely that we'll need to revert this change due to the following issues:

- [ ] #54254
- [ ] #75449

as well as issues around the console for non system tenants.

**Comment by ajstorm** [comment]
Yes, that's the meaningful one.  Currently we don't expose a console for non-system tenants, which is a significant regression which we're working with product on how to solve.  More generally though, I don't think we have high confidence that there aren't other paper-cuts which we've not yet uncovered, that will annoy users of `demo`.

**Comment by ajstorm** [comment]
To @ajwerner's point above, the lack of a console for non-system SQL servers is now tracked here: https://github.com/cockroachdb/cockroach/issues/78336

**Comment by ajstorm** [comment]
Removing the release blocker flag for 22.1 now that we've merged #78168.

**Comment by rmloveland** [comment]
Quoth @ajstorm :

>  More generally though, I don't think we have high confidence that there aren't other paper-cuts which we've not yet uncovered, that will annoy users of demo.

We just discussed this in the docs team weekly meeting.  As of this writing `demo` having MT enabled is resulting in the following issues that make us desire it to be disabled for `demo` in the v22.2 release:

- The console output at startup says a bunch of stuff about "system tenant", "tenant 1", which is not meaningful (and in fact confusing) to a `demo` user, since [`demo` is billed as "a temporary, in-memory CockroachDB cluster"](https://www.cockroachlabs.com/docs/dev/cockroach-demo.html) so arguably it should not require understanding some internal implementation detail exists like multi-tenancy.
- (Experience from an internal docs team user) With MT enabled for `demo` by default, if a user connects to port 8081 (system tenant) they can't see any statements they run from the SQL client. If they connect to port 8080 (tenant 1) they get an error in DB Console for some of the screens due to https://github.com/cockroachdb/cockroach/issues/80789
  -  the [`demo` docs](https://www.cockroachlabs.com/docs/dev/cockroach-demo.html) do say "This command is therefore recommended only as an easy way to experiment with the CockroachDB SQL dialect", so maybe that's OK.  But as y'all probably know `demo` has grown a lot of flags/features, and a lot of internal folks use it for more than just "try out a SQL query", so perhaps that line needs to be struck from the docs, arguably.  (Def open to removing that line from the docs pending results of this convo)

summary: Enabling MT for `demo` by default results in some UX regressions for end users in its current state.  I hope we will consider turning it off for the released v22.2 binary unless/until it becomes "exactly the same" from a user POV (which it is not right now, as far as we can tell)

what do you all think?

**Comment by rmloveland** [comment]
Another issue discovered while working on [DOC-3353](https://cockroachlabs.atlassian.net/browse/DOC-3353): starting `demo` as described in our [local multi-region TPCC instructions](https://www.cockroachlabs.com/docs/v22.1/performance-benchmarking-with-tpcc-local-multiregion.html) using on the latest `v22.2.0-alpha.1` release takes approximately 90 seconds on my work MBP

Meanwhile, with `--multitenant=false` it takes approx. 2 seconds (!)

Here are the commands for reference. Slow multitenant version:

```
cockroach demo --global --nodes 9 --no-example-database --insecure
```

Lightning fast old school version:

```
cockroach demo --multitenant=false --global --nodes 9 --no-example-database --insecure
```

**Comment by knz** [comment]
I vote to have `demo` run with `--multitenant=false` in 22.2 by default.
Then also keep it to `true` in the `master` branch for our internal development.

**Comment by ajstorm** [comment]
> I vote to have demo run with --multitenant=false in 22.2 by default.
Then also keep it to true in the master branch for our internal development.

Yes, we discussed this in our weekly a few weeks ago.  Sorry for not updating this issue then, and for causing @rmloveland to write such a long post arguing for something which we'd already agreed upon.  I'll make this change sometime this week.

**Comment by rmloveland** [comment]
Adam Storm ***@***.***> writes:

>     I vote to have demo run with --multitenant=false in 22.2 by
>     default.
>     Then also keep it to true in the master branch for our internal
>     development.
>
> Yes, we discussed this in our weekly a few weeks ago. Sorry for not
> updating this issue then, and for causing @rmloveland to write such a
> long post arguing for something which we'd already agreed upon. I'll
> make this change sometime this week.

Awesome, thanks @ajstorm! And thanks also @knz!

FWIW it was useful for me to spend some time checking this against a
couple of things in our docs, so no apologies needed

**Comment by ajstorm** [comment]
Addressed for 22.2 with #72341

**Comment by mgartner** [comment]
Enabling multi-tenancy in the demo still significantly impacts performance, as seen in #98245. Based on the conversation above, it looks like we the demo's default was `--multitenant=false` in v22.2 in part because of this performance impact. What has changed since those discussions that makes this performance regression acceptable now in v23.1?

**Comment by knz** [comment]
1) today at team meeting we've discussed the opportunity to have the _release_ version of cockroach demo run without multitenancy, but keep it enabled on the `master` branch for use by our team. This is what we did in the last 2 releases and at this point we're likely to do the same again.

2) to the performance problem in particular: yes we know of this, and it's related to the throttling performed on secondary tenants (also applicable in serverless). There's a couple of PRs in the pipeline to improve this.

**Comment by ajstorm** [comment]
With https://github.com/cockroachdb/cockroach/pull/99483 merged, I'm removing the release blocker tag for 23.1.  We can revisit (again) in 23.2.

**Comment by ajstorm** [comment]
I merged https://github.com/cockroachdb/cockroach/pull/112404 today which disabled this in 23.2.  Leaving this issue open so that we can revisit in 24.1.  We're getting much closer, but we're not all the way there just yet.
